cpp/server/ct-server.cc

```diff
@@ -237,11 +237,22 @@ void CleanUpEntries(ConsistentStore<LoggedCertificate>* store,
 
   while (true) {
     if (election->IsMaster()) {
-      const ScopedLatency sequencer_delete_latency(
-          sequencer_delete_latency_ms.ScopedLatency());
-      util::Status status(store->CleanupOldEntries());
-      if (!status.ok()) {
-        LOG(WARNING) << "Problem cleaning up old entries: " << status;
+      // Keep cleaning up until there's no more work to do.
+      // This should help to keep the etcd contents size down during heavy
+      // load.
+      while (true) {
+        const ScopedLatency sequencer_delete_latency(
+            sequencer_delete_latency_ms.ScopedLatency());
```

Comment:
Yeah, fair point - there's still the op latency inside the method itself, which is probably more useful.